File: fel-adapter/src/de/proskor/fel/impl/EventTypeImpl.scala
package de.proskor.fel.impl

import java.util.{List => JavaList}

import scala.collection.JavaConversions.seqAsJavaList

import de.proskor.automation.Element
import de.proskor.fel.container.EventTypeContainer
import de.proskor.fel.event.EventInstance
import de.proskor.fel.event.EventType

class EventTypeImpl(private[impl] val peer: Element) extends EntityImpl(peer) with EventType {
  override def getContainer: EventTypeContainer = {
    val types = for {
      connector <- peer.connectors
      if connector.source == peer && connector.stereotype == "belongsTo"
      target = connector.target
    } yield new EventTypeContainerImpl(target)
    types.headOption.orNull
  }

  override def getInstances: JavaList[EventInstance] = {
    val instances = for {
      connector <- peer.connectors
      if connector.target == peer && connector.stereotype == "instanceOf"
      source = connector.source
    } yield new EventInstanceImpl(source)
    instances.toList
  }

  override def addInstance(instance: EventInstance) {}
}

File: ea-fel/src/de/proskor/fel/stub/EventTypeImpl.java
package de.proskor.fel.stub;

import java.util.ArrayList;
import java.util.List;

import de.proskor.fel.container.EventTypeContainer;
import de.proskor.fel.event.EventInstance;
import de.proskor.fel.event.EventType;

public class EventTypeImpl extends EntityImpl implements EventType {
	private EventTypeContainer container;
	private List<EventInstance> instances = new ArrayList<EventInstance>();

	@Override
	public EventTypeContainer getContainer() {
		return this.container;
	}

	public void setContainer(EventTypeContainer container) {
		this.container = container;
	}

	@Override
	public List<EventInstance> getInstances() {
		return this.instances;
	}

	@Override
	public void addInstance(EventInstance eventInstance) {
		this.instances.add(eventInstance);
		eventInstance.setType(this);
	}
	
	@Override
	public String getQualifiedName() {
		return getContainer().getQualifiedName() + "." + getName();
	}
}
